src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs

```diff
@@ -92,6 +92,31 @@ protected string BuildRedirectUri(string targetPath)
             return Request.Scheme + "://" + Request.Host + OriginalPathBase + targetPath;
         }
 
+        /// <summary>
+        /// Handles errors by redirecting to an error handler path if configured.  Otherwise sets the status code to 500.
+        /// </summary>
+        /// <param name="error"></param>
+        /// <returns></returns>
+        protected virtual Task HandleErrorAsync(ErrorContext error)
```

Comment:
Yeah that's definitely one issue right now and related to the greater problem is a lot of the errors are in HandleAuthenticate which currently returns a Ticket and has no flow control...  So I couldn't use this new error behavior there at all